### PR TITLE
Feature/update dev build

### PR DESCRIPTION
**Pull request by genie9**
### Description

This commit adds the use of .env file for `docker-compose up` so that developer's own environment variables would not be accidentally pushed to the repository. Also makes it easy to handle your own development environment.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (this needs a follow up PR)

### Changes Made
Docker-compose and docker-compose-tls files changed to use variables from .env file. 
Adds .env.example file to copy it to user's local .env file. 
Updates README with instructions.
Updates .gitignore to ignore .env file
Updates github workflow commands on docker-compose with env file flag

Additional changes:
Update README with instructions on Python requirements.
Update pre-commit script to execute tests even if .tox folder does not exist. 

### Testing

- [ ] Unit Tests
- [ ] Integration Tests
- [x] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

**Review by blankdots — COMMENTED** [review]
Looks good, I think we might want to do the same kind of treatment to `docker-compose-tls.yml` to support `.env` file

For github actions we might need to add:
- ` --env-file .env.example` to  https://github.com/CSCfi/metadata-submitter/blob/develop/.github/workflows/int.yml#L29 and https://github.com/CSCfi/metadata-submitter/blob/develop/.github/workflows/int.yml#L58

**Comment by blankdots** [comment]
this might also imply changes in the https://github.com/CSCfi/metadata-submitter-frontend/blob/master/.github/workflows/e2etests.yml#L38 line so before we merge this we need to update that as well

File: .github/workflows/int.yml [hunk 1]
```diff
@@ -26,8 +26,8 @@ jobs:
         
     - name: Start Services
       run: |
-        docker-compose up -d --build
-        sleep 30
+        docker-compose up -d --build --env-file .env.example
```
Comment (blankdots) [inline .github/workflows/int.yml:29]:
```suggestion
        docker-compose --env-file .env.example up -d --build
```
the file s envhould come first as per docker-compose syntax

File: .github/workflows/int.yml [hunk 2]
```diff
@@ -55,8 +55,8 @@ jobs:
 
     - name: Start services with TLS enabled
       run: |
-        docker-compose -f docker-compose-tls.yml up -d
-        sleep 30
+        docker-compose -f docker-compose-tls.yml up -d --env-file .env.example
```
Comment (blankdots) [inline .github/workflows/int.yml:58]:
```suggestion
        docker-compose -f docker-compose-tls.yml --env-file .env.example up -d --build
```

**Review by blankdots — COMMENTED** [review]
also `docker-compose-tls,yml` file should have similar environment variables as `docker-compose.yml`

File: .github/workflows/int.yml (hunk 2 above)
Comment (genie9) [reply .github/workflows/int.yml]: thanks!

**Review by blankdots — APPROVED** [review]

**Pull request merged**